src/lib/externaldata.py

```diff
@@ -67,14 +78,15 @@ class State(Enum):
 
     @classmethod
     def from_source(cls, source_path: str, source: t.Dict) -> t.Optional[ExternalBase]:
-        if not source.get("url"):
-            return None
-
         try:
-            data_type = cls.Type(source.get("type"))
-        except ValueError:
+            jsonschema.validate(source, cls.SOURCE_SCHEMA)
+        except jsonschema.ValidationError as err:
+            raise ManifestLoadError("Error reading source") from err
+
+        if not source.get("url"):
             return None
 
+        data_type = cls.Type(source["type"])
```

Comment:
Previously, an unrecognised `type` would be caught by `except ValueError`. With this change the `ValueError` will propagate outwards. I don't think we want that because there are many other valid types beyond the four in `class Type(Enum)` above.

Reply:
I see, this wasn't intentional. Redone this wrapping unsupported source types in custom exception, handled in `MenfestChecker` instead of `ExternalData`.